task-launcher/src/tasks/shared/trials/afcStimulus.js

```diff
@@ -23,31 +23,55 @@ let currPracticeChoiceMix = [];
 let currPracticeAnswerIdx;
 let trialsOfCurrentType = 0;
 
-let audioSource;
 let keyboardResponseMap = {};
 // Only used for keyboard responses
 let startTime;
 const incorrectPracticeResponses = [];
 
-const playAudio = async (audioUri) => {
-  const jsPsychAudioCtx = jsPsych.pluginAPI.audioContext();
+
+// Enum for the side the stimulus is shown on and also for button_choices
+const ExtraAudioSourceKeys = Object.freeze({
+  Replay: 'Replay',
+  ButtonFeedback: 'ButtonFeedback',
+  KeyFeedback: 'KeyFeedback',
+  StaggeredButton: 'StaggeredButton',
+});
+
+async function playAudioAsync(audioUri, jsPsychPluginAPI, extraAudioSources, extraAudioSourceKey) {
+  // Don't play audio if it's already playing
+  if (extraAudioSources[extraAudioSourceKey]) {
+    console.info(`Audio is already playing for ${extraAudioSourceKey}`);
+    return;
+  }
+
   // Returns a promise of the AudioBuffer of the preloaded file path.
-  const audioBuffer = await jsPsych.pluginAPI.getAudioBuffer(audioUri);
+  const audioBuffer = await jsPsychPluginAPI.getAudioBuffer(audioUri);
+  const jsPsychAudioCtx = jsPsychPluginAPI.audioContext();
   const audioSource = jsPsychAudioCtx.createBufferSource();
   audioSource.buffer = audioBuffer;
   audioSource.connect(jsPsychAudioCtx.destination);
   audioSource.start(0);
+  extraAudioSources[extraAudioSourceKey] = audioSource;
+  audioSource.onended = () => {
+    if (extraAudioSources[extraAudioSourceKey]) {
+      extraAudioSources[extraAudioSourceKey] = null;
+    }
+  };
 };
 
-const showStaggeredBtnAndPlaySound = (btn) => {
+function showStaggeredBtnAndPlaySound(btn, jsPsychPluginApi, extraAudioSources) {
```

Comment:
I personally prefer constants instead of function declarations because in JS, functions are hoisted which can lead to bad practices of calling functions before it is defined etc.